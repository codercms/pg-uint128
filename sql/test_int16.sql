-- Testing int16

-- Ops block

SELECT 1::int16 = 1::int16;
SELECT 1::int16 = 0::int16;
SELECT 1::int16 = 1::int2;
SELECT 1::int16 = 0::int2;
SELECT 1::int2 = 1::int16;
SELECT 1::int2 = 0::int16;
SELECT 1::int16 = 1::int4;
SELECT 1::int16 = 0::int4;
SELECT 1::int4 = 1::int16;
SELECT 1::int4 = 0::int16;
SELECT 1::int16 = 1::int8;
SELECT 1::int16 = 0::int8;
SELECT 1::int8 = 1::int16;
SELECT 1::int8 = 0::int16;

SELECT 1::int16 <> 0::int16;
SELECT 0::int16 <> 0::int16;
SELECT 1::int16 <> 0::int2;
SELECT 0::int16 <> 0::int2;
SELECT 1::int2 <> 0::int16;
SELECT 0::int2 <> 0::int16;
SELECT 1::int16 <> 0::int4;
SELECT 0::int16 <> 0::int4;
SELECT 1::int4 <> 0::int16;
SELECT 0::int4 <> 0::int16;
SELECT 1::int16 <> 0::int8;
SELECT 0::int16 <> 0::int8;
SELECT 1::int8 <> 0::int16;
SELECT 0::int8 <> 0::int16;

SELECT 1::int16 > 0::int16;
SELECT 0::int16 > 1::int16;
SELECT 1::int16 > 0::int2;
SELECT 0::int16 > 1::int2;
SELECT 1::int2 > 0::int16;
SELECT 0::int2 > 1::int16;
SELECT 1::int16 > 0::int4;
SELECT 0::int16 > 1::int4;
SELECT 1::int4 > 0::int16;
SELECT 0::int4 > 1::int16;
SELECT 1::int16 > 0::int8;
SELECT 0::int16 > 1::int8;
SELECT 1::int8 > 0::int16;
SELECT 0::int8 > 1::int16;

SELECT 0::int16 < 1::int16;
SELECT 1::int16 < 0::int16;
SELECT 0::int16 < 1::int2;
SELECT 1::int16 < 0::int2;
SELECT 0::int2 < 1::int16;
SELECT 1::int2 < 0::int16;
SELECT 0::int16 < 1::int4;
SELECT 1::int16 < 0::int4;
SELECT 0::int4 < 1::int16;
SELECT 1::int4 < 0::int16;
SELECT 0::int16 < 1::int8;
SELECT 1::int16 < 0::int8;
SELECT 0::int8 < 1::int16;
SELECT 1::int8 < 0::int16;

SELECT 1::int16 >= 0::int16;
SELECT 0::int16 >= 1::int16;
SELECT 1::int16 >= 1::int16;
SELECT 1::int16 >= 0::int2;
SELECT 0::int16 >= 1::int2;
SELECT 1::int16 >= 1::int2;
SELECT 1::int2 >= 0::int16;
SELECT 0::int2 >= 1::int16;
SELECT 1::int2 >= 1::int16;
SELECT 1::int16 >= 0::int4;
SELECT 0::int16 >= 1::int4;
SELECT 1::int16 >= 1::int4;
SELECT 1::int4 >= 0::int16;
SELECT 0::int4 >= 1::int16;
SELECT 1::int4 >= 1::int16;
SELECT 1::int16 >= 0::int8;
SELECT 0::int16 >= 1::int8;
SELECT 1::int16 >= 1::int8;
SELECT 1::int8 >= 0::int16;
SELECT 0::int8 >= 1::int16;
SELECT 1::int8 >= 1::int16;

SELECT 0::int16 <= 1::int16;
SELECT 1::int16 <= 0::int16;
SELECT 1::int16 <= 1::int16;
SELECT 0::int16 <= 1::int2;
SELECT 1::int16 <= 0::int2;
SELECT 1::int16 <= 1::int2;
SELECT 0::int2 <= 1::int16;
SELECT 1::int2 <= 0::int16;
SELECT 1::int2 <= 1::int16;
SELECT 0::int16 <= 1::int4;
SELECT 1::int16 <= 0::int4;
SELECT 1::int16 <= 1::int4;
SELECT 0::int4 <= 1::int16;
SELECT 1::int4 <= 0::int16;
SELECT 1::int4 <= 1::int16;
SELECT 0::int16 <= 1::int8;
SELECT 1::int16 <= 0::int8;
SELECT 1::int16 <= 1::int8;
SELECT 0::int8 <= 1::int16;
SELECT 1::int8 <= 0::int16;
SELECT 1::int8 <= 1::int16;

SELECT 120::int16 + 10::int16;
SELECT 120::int16 + 10::int2;
SELECT 170141183460469231731687303715884105727::int16 + 1::int2;
SELECT 120::int2 + 10::int16;
SELECT 32767::int2 + 1::int16;
SELECT 120::int16 + 10::int4;
SELECT 170141183460469231731687303715884105727::int16 + 1::int4;
SELECT 120::int4 + 10::int16;
SELECT 2147483647::int4 + 1::int16;
SELECT 120::int16 + 10::int8;
SELECT 170141183460469231731687303715884105727::int16 + 1::int8;
SELECT 120::int8 + 10::int16;
SELECT 9223372036854775807::int8 + 1::int16;

SELECT 120::int16 - 10::int16;
SELECT 120::int16 - 10::int2;
SELECT (-170141183460469231731687303715884105728)::int16 - 1::int2;
SELECT 120::int2 - 10::int16;
SELECT (-32768)::int2 - 1::int16;
SELECT 120::int16 - 10::int4;
SELECT (-170141183460469231731687303715884105728)::int16 - 1::int4;
SELECT 120::int4 - 10::int16;
SELECT (-2147483648)::int4 - 1::int16;
SELECT 120::int16 - 10::int8;
SELECT (-170141183460469231731687303715884105728)::int16 - 1::int8;
SELECT 120::int8 - 10::int16;
SELECT (-9223372036854775808)::int8 - 1::int16;

SELECT 120::int16 * 10::int16;
SELECT 120::int16 * 10::int2;
SELECT 170141183460469231731687303715884105727::int16 * 2::int2;
SELECT 120::int2 * 10::int16;
SELECT 32767::int2 * 2::int16;
SELECT 120::int16 * 10::int4;
SELECT 170141183460469231731687303715884105727::int16 * 2::int4;
SELECT 120::int4 * 10::int16;
SELECT 2147483647::int4 * 2::int16;
SELECT 120::int16 * 10::int8;
SELECT 170141183460469231731687303715884105727::int16 * 2::int8;
SELECT 120::int8 * 10::int16;
SELECT 9223372036854775807::int8 * 2::int16;

SELECT 120::int16 / 10::int16;
SELECT 120::int16 / 10::int2;
SELECT 1::int16 / 0::int2;
SELECT 120::int2 / 10::int16;
SELECT 1::int2 / 0::int16;
SELECT 120::int16 / 10::int4;
SELECT 1::int16 / 0::int4;
SELECT 120::int4 / 10::int16;
SELECT 1::int4 / 0::int16;
SELECT 120::int16 / 10::int8;
SELECT 1::int16 / 0::int8;
SELECT 120::int8 / 10::int16;
SELECT 1::int8 / 0::int16;

SELECT 120::int16 % 10::int16;
SELECT 120::int16 % 10::int2;
SELECT 1::int16 % 0::int2;
SELECT 120::int2 % 10::int16;
SELECT 1::int2 % 0::int16;
SELECT 120::int16 % 10::int4;
SELECT 1::int16 % 0::int4;
SELECT 120::int4 % 10::int16;
SELECT 1::int4 % 0::int16;
SELECT 120::int16 % 10::int8;
SELECT 1::int16 % 0::int8;
SELECT 120::int8 % 10::int16;
SELECT 1::int8 % 0::int16;

