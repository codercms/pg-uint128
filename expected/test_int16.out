-- Testing int16
-- Ops block
SELECT 1::int16 = 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 = 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 = 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 = 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 = 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 = 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 = 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 = 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 = 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 = 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 = 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 = 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 = 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 = 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <> 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <> 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <> 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <> 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 <> 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 <> 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <> 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <> 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 <> 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 <> 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <> 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <> 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 <> 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 <> 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 > 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 > 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 > 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 > 1::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 > 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 > 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 > 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 > 1::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 > 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 > 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 > 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 > 1::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 > 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 > 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 0::int16 < 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 < 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 0::int16 < 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 < 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 0::int2 < 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 < 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 0::int16 < 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 < 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 0::int4 < 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 < 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 0::int16 < 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 < 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 0::int8 < 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 < 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 >= 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 >= 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 >= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 >= 0::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 >= 1::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 >= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 >= 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 >= 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 >= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 >= 0::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 >= 1::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 >= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 >= 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 >= 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 >= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 >= 0::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 >= 1::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 >= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 >= 0::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 >= 1::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 >= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 <= 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 <= 0::int2;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <= 1::int2;
 ?column? 
----------
 t
(1 row)

SELECT 0::int2 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int2 <= 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int2 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 <= 0::int4;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <= 1::int4;
 ?column? 
----------
 t
(1 row)

SELECT 0::int4 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int4 <= 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int4 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 0::int16 <= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 1::int16 <= 0::int8;
 ?column? 
----------
 f
(1 row)

SELECT 1::int16 <= 1::int8;
 ?column? 
----------
 t
(1 row)

SELECT 0::int8 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 1::int8 <= 0::int16;
 ?column? 
----------
 f
(1 row)

SELECT 1::int8 <= 1::int16;
 ?column? 
----------
 t
(1 row)

SELECT 120::int16 + 10::int16;
 ?column? 
----------
      130
(1 row)

SELECT 120::int16 + 10::int2;
 ?column? 
----------
      130
(1 row)

SELECT 170141183460469231731687303715884105727::int16 + 1::int2;
ERROR:  int16 out of range
SELECT 120::int2 + 10::int16;
 ?column? 
----------
      130
(1 row)

SELECT 32767::int2 + 1::int16;
ERROR:  int2 out of range
SELECT 120::int16 + 10::int4;
 ?column? 
----------
      130
(1 row)

SELECT 170141183460469231731687303715884105727::int16 + 1::int4;
ERROR:  int16 out of range
SELECT 120::int4 + 10::int16;
 ?column? 
----------
      130
(1 row)

SELECT 2147483647::int4 + 1::int16;
ERROR:  int4 out of range
SELECT 120::int16 + 10::int8;
 ?column? 
----------
      130
(1 row)

SELECT 170141183460469231731687303715884105727::int16 + 1::int8;
ERROR:  int16 out of range
SELECT 120::int8 + 10::int16;
 ?column? 
----------
      130
(1 row)

SELECT 9223372036854775807::int8 + 1::int16;
ERROR:  int8 out of range
SELECT 120::int16 - 10::int16;
 ?column? 
----------
      110
(1 row)

SELECT 120::int16 - 10::int2;
 ?column? 
----------
      110
(1 row)

SELECT (-170141183460469231731687303715884105728)::int16 - 1::int2;
ERROR:  int16 out of range
SELECT 120::int2 - 10::int16;
 ?column? 
----------
      110
(1 row)

SELECT (-32768)::int2 - 1::int16;
ERROR:  int2 out of range
SELECT 120::int16 - 10::int4;
 ?column? 
----------
      110
(1 row)

SELECT (-170141183460469231731687303715884105728)::int16 - 1::int4;
ERROR:  int16 out of range
SELECT 120::int4 - 10::int16;
 ?column? 
----------
      110
(1 row)

SELECT (-2147483648)::int4 - 1::int16;
ERROR:  int4 out of range
SELECT 120::int16 - 10::int8;
 ?column? 
----------
      110
(1 row)

SELECT (-170141183460469231731687303715884105728)::int16 - 1::int8;
ERROR:  int16 out of range
SELECT 120::int8 - 10::int16;
 ?column? 
----------
      110
(1 row)

SELECT (-9223372036854775808)::int8 - 1::int16;
ERROR:  int8 out of range
SELECT 120::int16 * 10::int16;
 ?column? 
----------
     1200
(1 row)

SELECT 120::int16 * 10::int2;
 ?column? 
----------
     1200
(1 row)

SELECT 170141183460469231731687303715884105727::int16 * 2::int2;
ERROR:  int16 out of range
SELECT 120::int2 * 10::int16;
 ?column? 
----------
     1200
(1 row)

SELECT 32767::int2 * 2::int16;
ERROR:  int2 out of range
SELECT 120::int16 * 10::int4;
 ?column? 
----------
     1200
(1 row)

SELECT 170141183460469231731687303715884105727::int16 * 2::int4;
ERROR:  int16 out of range
SELECT 120::int4 * 10::int16;
 ?column? 
----------
     1200
(1 row)

SELECT 2147483647::int4 * 2::int16;
ERROR:  int4 out of range
SELECT 120::int16 * 10::int8;
 ?column? 
----------
     1200
(1 row)

SELECT 170141183460469231731687303715884105727::int16 * 2::int8;
ERROR:  int16 out of range
SELECT 120::int8 * 10::int16;
 ?column? 
----------
     1200
(1 row)

SELECT 9223372036854775807::int8 * 2::int16;
ERROR:  int8 out of range
SELECT 120::int16 / 10::int16;
 ?column? 
----------
       12
(1 row)

SELECT 120::int16 / 10::int2;
 ?column? 
----------
       12
(1 row)

SELECT 1::int16 / 0::int2;
ERROR:  division by zero
SELECT 120::int2 / 10::int16;
 ?column? 
----------
       12
(1 row)

SELECT 1::int2 / 0::int16;
ERROR:  division by zero
SELECT 120::int16 / 10::int4;
 ?column? 
----------
       12
(1 row)

SELECT 1::int16 / 0::int4;
ERROR:  division by zero
SELECT 120::int4 / 10::int16;
 ?column? 
----------
       12
(1 row)

SELECT 1::int4 / 0::int16;
ERROR:  division by zero
SELECT 120::int16 / 10::int8;
 ?column? 
----------
       12
(1 row)

SELECT 1::int16 / 0::int8;
ERROR:  division by zero
SELECT 120::int8 / 10::int16;
 ?column? 
----------
       12
(1 row)

SELECT 1::int8 / 0::int16;
ERROR:  division by zero
SELECT 120::int16 % 10::int16;
 ?column? 
----------
        0
(1 row)

SELECT 120::int16 % 10::int2;
 ?column? 
----------
        0
(1 row)

SELECT 1::int16 % 0::int2;
ERROR:  division by zero
SELECT 120::int2 % 10::int16;
 ?column? 
----------
        0
(1 row)

SELECT 1::int2 % 0::int16;
ERROR:  division by zero
SELECT 120::int16 % 10::int4;
 ?column? 
----------
        0
(1 row)

SELECT 1::int16 % 0::int4;
ERROR:  division by zero
SELECT 120::int4 % 10::int16;
 ?column? 
----------
        0
(1 row)

SELECT 1::int4 % 0::int16;
ERROR:  division by zero
SELECT 120::int16 % 10::int8;
 ?column? 
----------
        0
(1 row)

SELECT 1::int16 % 0::int8;
ERROR:  division by zero
SELECT 120::int8 % 10::int16;
 ?column? 
----------
        0
(1 row)

SELECT 1::int8 % 0::int16;
ERROR:  division by zero
